webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -0,0 +1,150 @@
+        .map((workspace) =>
+          getWorkspaceProjects(
+            { id: workspace.id, sortBy: 'SORT_BY_NAME' },
+            { signal: canceler.current.signal },
+          ));
+      const projectResponse = (await Promise.all(projectAPIList))
+        .map((project) => project.projects);
+
+      // Promise.all preserves the order
+      const tempWorkspaceMap: Map<Workspace, Project[]> = new Map();
+      for (const workspace of filteredWorkspaces.reverse()) {
+        const projects = projectResponse.pop();
+        tempWorkspaceMap.set(workspace, projects ?? []);
+      }
+      setWorkspaceMap(tempWorkspaceMap);
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Please try again later.',
+        publicSubject: 'Unable to fetch data.',
+        silent: false,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [ ]);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchText(e.target.value);
+  };
+
+  const onShowModal = () => {
+    setIsModalVisible(true);
+    fetchData();
+  };
+
+  const onHideModal = () => setIsModalVisible(false);
+
+  const onClickProject = useCallback((project: Project) => {
+    routeToReactUrl(paths.projectDetails(project.id));
+    onHideModal();
+  }, []);
+
+  const onClickWorkspace = useCallback((workspaceId: number) => {
+    routeToReactUrl(paths.workspaceDetails(workspaceId));
+    onHideModal();
+  }, []);
+
+  const treeData: DefaultOptionType[] = useMemo(() => {
+    const data: DefaultOptionType[] = Array.from(workspaceMap)
+      .map(([ workspace, projects ]) => {
+        const treeChildren: DefaultOptionType[] = projects
+          .filter((project) => project.name.includes(searchText))
+          .map((project) => ({
+            title: (
+              <div className={`${css.flexRow} ${css.ellipsis}`}>
+                <Icon name="experiment" />
+                <Link onClick={() => onClickProject(project)}>{project.name}</Link>
+              </div>
+            ),
+            value: `project-${project.id}`,
+          }));
+        return ({
+          children: treeChildren,
+          title: (
+            <div className={`${css.flexRow} ${css.ellipsis}`}>
+              <Icon name="workspaces" />
+              <Link onClick={() => onClickWorkspace(workspace.id)}>{workspace.name}</Link>
+            </div>
+          ),
+          value: `workspace-${workspace.id}`,
+        });
+      })
+      .filter((item) => searchText.length > 0 ? item.children.length > 0 : true);
+    return data;
+  }, [ onClickProject, onClickWorkspace, searchText, workspaceMap ]);
+
+  return (
+    <div>
+      <div onClick={onShowModal}>
+        {children}
+      </div>
+      <Modal
+        closable={false}
+        footer={null}
+        title={(
+          <Input
+            placeholder="Search and Jump to workspace or project"
```

Comment:
"Search projects..." or similar

also if you pass the input a ref, you can call `focus()` on it with a use effect to put the cursor there by default.

Reply:
yes added